lib/plugins/aws/deploy/lib/ensure-valid-bucket-exists.js

```diff
@@ -111,51 +112,81 @@ module.exports = {
       stackTags = Object.assign(stackTags, this.serverless.service.provider.stackTags);
     }
 
-    const params = {
+    const createChangeSetParams = {
       StackName: stackName,
+      ChangeSetName: changeSetName,
+      ChangeSetType: 'UPDATE',
       Capabilities: ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
       Parameters: [],
       Tags: Object.keys(stackTags).map((key) => ({ Key: key, Value: stackTags[key] })),
       TemplateBody: JSON.stringify(templateBody),
     };
 
+    const executeChangeSetParams = {
+      StackName: stackName,
+      ChangeSetName: changeSetName,
+    };
+
     const customDeploymentRole = this.provider.getCustomDeploymentRole();
     if (customDeploymentRole) {
-      params.RoleARN = customDeploymentRole;
+      createChangeSetParams.RoleARN = customDeploymentRole;
```

Comment:
I think in context of this PR, it was supposed to be improved, so logic is not repeated (if I see correctly we're doing same thing when creating stack for first time)?

https://github.com/serverless/serverless/pull/10317#discussion_r763846341

Reply:
As discussed on Slack, it will be added as separate PR